src/agent/command_handler/include/command_handler.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <boost/asio.hpp>
+#include <command_store.hpp>
+
+#include <iostream>
+#include <optional>
+#include <queue>
+
+namespace command_handler
+{
+    class CommandHandler
+    {
+    public:
+        CommandHandler();
+
+        template<typename T>
```

Comment:
Why isn't this function definition in the `.cpp`? Is the `.cpp` even necessary as it is almost empty?

Reply:
The function needs to be in the hpp since it's a template function. The cpp is not really needed, so... gone.